### ISSUE TITLE
Segmentation Fault

### ISSUE THREAD

**Issue by wwderw** [RandyGaul/cute_framework#99]
OS: Linux
G++: 12.2.1

==============Copy/Paste from Discord====================
On Linux, I have built the cute framework using CMake and brought it in where I need it to.  Everything builds when I try to do a simple  app example, so I think that part is good.  Everything links like it should, but running the App example on the website, I get a segmentation fault.  I have run it as it is on the site and I have run it changing the context from just the default to the OpenGL one.  Both build fine, no errors, just that fault when trying to run.  Nothing seems to come up with gdb.

Original Code:
>#include <cute.h>
using namespace Cute;
int main(int argc, char* argv[])
{
    // Create a window with a resolution of 640 x 480.
    int options = APP_OPTIONS_DEFAULT_GFX_CONTEXT | APP_OPTIONS_WINDOW_POS_CENTERED;
    Result result = make_app("Fancy Window Title", 0, 0, 640, 480, options, argv[0]);
    if (is_error(result)) return -1;
    while (app_is_running())
    {
        app_update();
        // All your game logic and updates go here...
        app_draw_onto_screen();
    }
    destroy_app();
    return 0;
}

And with the one change that I did:

> #include <cute.h>
using namespace Cute;
int main(int argc, char* argv[])
{
    // Create a window with a resolution of 640 x 480.
    int options = APP_OPTIONS_OPENGL_CONTEXT | APP_OPTIONS_WINDOW_POS_CENTERED;
    Result result = make_app("Fancy Window Title", 0, 0, 640, 480, options, argv[0]);
    if (is_error(result)) return -1;
    while (app_is_running())
    {
        app_update();
        // All your game logic and updates go here...
        app_draw_onto_screen();
    }
    destroy_app();
    return 0;
}

**Comment by RandyGaul** [comment]
One thing that comes to mind is try and make sure you have OpenGL installed on your Linux machine. Something like this:

```
sudo apt-get update -qq
sudo apt-get install gcc-multilib
sudo apt-get install -y --no-install-recommends libglfw3 libglfw3-dev libx11-dev libxcursor-dev libxrandr-dev libxinerama-dev libxi-dev libxext-dev libxfixes-dev
```

**Comment by wwderw** [comment]
Confirmed that everything is as it should be.

**Comment by RandyGaul** [comment]
Okay so I took a look and wasn't able to reproduce any issues. I was actually able to run all the samples and tests successfully through WSL2 on Windows. My best guess is you ran into an audio initialization problem. Unfortunately I had a simple assert there instead of return a `CF_Result` with any error details. I've fixed this here https://github.com/RandyGaul/cute_framework/commit/3c98fc4312d2ca14580a5febc8500e25be0ca3e1

I also updated a list of things to install on Linux to try:

```
sudo apt-get update -qq
sudo apt-get install build-essential gcc-multilib cmake
sudo apt-get install libasound2-dev libpulse-dev 
sudo apt-get install -y --no-install-recommends libglfw3 libglfw3-dev libx11-dev libxcursor-dev libxrandr-dev libxinerama-dev libxi-dev libxext-dev libxfixes-dev
```
